backends/printing/win32/win32-printman.cpp

```diff
@@ -109,6 +119,48 @@ void Win32PrintJob::setTextColor(int r, int g, int b) {
 	SetTextColor(hdcPrint, RGB(r, g, b));
 }
 
+Common::Rect Win32PrintJob::getTextBounds(const Common::String &text) const {
+	SIZE winSize;
+	GetTextExtentPointA(hdcPrint, text.c_str(), text.size(), &winSize);
+	return Common::Rect(winSize.cx, winSize.cy);
+}
+
+Common::Rational Win32PrintJob::getPixelAspectRatio() const {
```

Comment:
Do you have any use-case for all these parameters? E.g. how do you think it could used by engines and by which specifically?

Reply:
I don't have use cases for all of them. But the new `testbed` code shows some of them in use.

Reply:
Then please remove them as they are not used by real engines.

Reply:
The way I see it, this is a complicated matter.

A feature added now is a burden on all backends that want to implement this. But a feature *not* added now that turns out to be needed later is an even bigger challenge. Editing the interface by adding member functions will cause existing implementations to break. Breaking existing stuff is very uncool.

As such I've been using some guidelines for what does and does not make it:
* Is the feature portable? If win, ios and android apis all have it its much more likely to make it than if it is just one of them.
* Is the feature complex to implement? A simple getter that directly queries the platform api is much simpler than something that would require bringing in hundreds, if not thousands, of lines of code.
* Is there a reasonable default if it can't be implemented? Hardcoding a return value that makes sense is much better than having to call `error`.